### DISCUSSION TITLE
How to keep/lock filter

### DISCUSSION THREAD

**Discussion by tkkcc** [jarun/nnn#1691] [Show and tell]
There are many folders in this dir, I have this use case
1. filter with keywords, and move to one of them: ` /abc<esc>jj`
2. nagivate into it and navigate back: `lh`
3. reapply filter and move to the next one in filtered result: `/<ctrl-l><esc>jjj`

I usually repeat step 2 and 3 for several times, and it's a lot of typing.

Is there a way to keep/lock filter on current dir, so i can quickly browsing filtered result.

**Comment by lawnowner** [comment]
You can use the  [list mode in a plugin](https://www.github.com/jarun/nnn/tree/master/plugins#send-data-to-nnn) to keep one or more filters by selecting the files in each filtered directory and listing the selected files, e.g. for a single filter in the current directory: `/abc<esc>a;l`. Make a [tmp copy of the selection file](https://www.github.com/jarun/nnn/tree/master/plugins#send-data-to-nnn) so that it can be cleared in the plugin. Remove the tmp file at the end of the shell function or script used to launch nnn. Currently each nnn instance allows having [one](https://www.github.com/jarun/nnn/wiki/Troubleshooting#cant-list-within-list) list view.

```
sel=${NNN_SEL:-${XDG_CONFIG_HOME:-$HOME/.config}/nnn/.selection}
seltmp=${sel}tmp

if ! [ -p "$NNN_PIPE" ] || ! [ -s "$sel" ]; then
  exit 1
fi

cat "$sel" > "$seltmp"
printf "-" > "$NNN_PIPE"

printf '%s' "+l" > "$NNN_PIPE"
cat "$seltmp" > "$NNN_PIPE"
```
